R/as-data-frame.R

```diff
@@ -73,3 +73,49 @@ as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE
     names(out) <- names(x)
     return(structure(out, class="data.frame", row.names=c(NA, -nrow(x))))
 }
+
+
+#' as.data.frame method for VariableCatalog
+#'
+#' This method allows you to import the VariableCatalog into your R sessions in 
+#' order to facilliate interactive use. Modifying the dataframe produced by this
+#' method will not update the Crunch web app, for information about modifying 
+#' variables in the app see: \code{vingette("variables",
+#' package = "crunch")}  
+#' 
+#' @param x
+#' A VariableCatalog produced by \code{variables(ds)}.
+#' @param ... Additional arguments passed to \code{data.frame}
+#' @param row.names part of as.data.frame signature. Ignored.
+#' @param optional part of as.data.frame signature. Ignored.
+#' @param fields 
+#' A character vector of the fields from the variable catalog which you would like 
+#' included in the datafame. To include all fields or see which fields are available
+#' set fields to "all". 
+#' @return A data frame
+#' @examples 
+#' \dontrun{
+#' ds <- loadDataset("iris")
+#' vars <- variables(ds)
+#' var_df <- as.data.frame(vars, fields = "all")
+#' }
+#' 
+#' @rdname VariableCatalog-to-Data-Frame
+#' @export
+as.data.frame.VariableCatalog <- function(x, 
+    row.names = NULL, 
+    optional = FALSE, 
+    ...,
+    fields = c("alias", "name", "type")) {
+    out <- catalogToDataFrame(x, rownames = NA)
```

Comment:
Let me remind myself first.... Oh I think it's that each variable catalog has different attributes, and we only want to throw an error if the field is not present in any of the variables in the dataset. So it seemed better to get all the available fields for each variable, and then check to see if the field names are valid.